core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -418,17 +419,50 @@ protected void flushRequests(Array<BuildRequest> requests){
                 if(copy.hasConfig && copy.block.posConfig){
                     copy.config = Pos.get(Pos.x(copy.config) + copy.x - copy.originalX, Pos.y(copy.config) + copy.y - copy.originalY);
                 }
+
+                copy.block = didyoumean(req.x, req.y, req.block, req.rotation);
+
                 player.addBuildRequest(copy);
             }
         }
     }
 
+    protected Block didyoumean(int x, int y, Block block, int rotation){
+        Tile over = world.ltile(x, y);
+
+        if(over.block() != null){
```

Comment:
Redundant condition, and again, `over.block()` **cannot be null**, only the tile itself can.